mycpp/error_types.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
```

Comment:
I think we should stick with the #include guards, it looks like it's not standard

https://stackoverflow.com/questions/23696115/is-pragma-once-part-of-the-c11-standard

In practice I'm not sure if it will ever show up, though maybe someone wants to build Oil on a weird embedded system with a weird C++ compiler (if that even exists -- I know there are weird C compilers, not sure about C++)

Reply:
Yeah I can change them.

Reply:
I've actually had these include guards bite me before, which is why I prefer `pragma once`, but it's pretty unlikely